scipy/diff/statsmodels/_derivative.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import division
+import numpy as np
+from _epsilon_generator import _epsilon
+
+
+def derivative(f, x, **options):
+    '''
+    Derivative of a function
+
+    Parameters
+    ----------
+    x : array
+        parameters at which the derivative is evaluated
+
+    f : function
```

Comment:
It took me a while to realize that I had things the wrong way around. Try:

    >>> derivative(np.arange(10), lambda x: x)

It gives ``TypeError: bad operand type for abs(): 'function'``. A more informative exception would be useful.